File: plugins/maven/src/test/java/org/jetbrains/idea/maven/project/actions/RemoveManagedFilesActionTest.java
// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package org.jetbrains.idea.maven.project.actions;

import com.intellij.maven.testFramework.MavenMultiVersionImportingTestCase;
import com.intellij.testFramework.TestActionEvent;
import org.junit.Test;

public class RemoveManagedFilesActionTest extends MavenMultiVersionImportingTestCase {
  @Test
  public void testUnlinkMavenProjectsOnlyVisibleForRootProjects() {
    var parentFile = createProjectPom("""
                  <groupId>group</groupId>
                  <artifactId>parent</artifactId>
                  <version>1</version>
                  <packaging>pom</packaging>
                  <modules>
                    <module>m1</module>
                  </modules>
                  """);
    var m1File = createModulePom("m1", """
                  <artifactId>m1</artifactId>
                  <version>1</version>
                  <parent>
                    <groupId>group</groupId>
                    <artifactId>parent</artifactId>
                  </parent>
                  """);
    importProject();

    var action = new RemoveManagedFilesAction();
    var parentActionVisible = action.isVisible(TestActionEvent.createTestEvent(action, createTestDataContext(parentFile)));
    var m1ActionVisible = action.isVisible(TestActionEvent.createTestEvent(action, createTestDataContext(m1File)));

    assertTrue(parentActionVisible);
    assertFalse(m1ActionVisible);
  }
}

File: plugins/maven/src/test/java/org/jetbrains/idea/maven/project/actions/RemoveManagedFilesActionTest.kt
// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package org.jetbrains.idea.maven.project.actions

import com.intellij.testFramework.TestActionEvent
import kotlinx.coroutines.runBlocking
import com.intellij.testFramework.junit5.TestApplication
import com.intellij.maven.testFramework.fixtures.MavenVersionArguments
import com.intellij.maven.testFramework.fixtures.createModulePom
import com.intellij.maven.testFramework.fixtures.createProjectPom
import com.intellij.maven.testFramework.fixtures.createTestDataContext
import com.intellij.maven.testFramework.fixtures.importProjectAsync
import com.intellij.maven.testFramework.fixtures.mavenImportingFixture
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedClass
import org.junit.jupiter.params.provider.ArgumentsSource

@TestApplication
@ParameterizedClass
@ArgumentsSource(MavenVersionArguments::class)
class RemoveManagedFilesActionTest(mavenVersion: String, modelVersion: String) {

  private val maven by mavenImportingFixture(
    mavenVersion = mavenVersion,
    modelVersion = modelVersion
  )
  
  @Test
  fun testUnlinkMavenProjectsOnlyVisibleForRootProjects() = runBlocking {
    val parentFile = maven.createProjectPom("""
                  <groupId>group</groupId>
                  <artifactId>parent</artifactId>
                  <version>1</version>
                  <packaging>pom</packaging>
                  <modules>
                    <module>m1</module>
                  </modules>
                  """.trimIndent())
    val m1File = maven.createModulePom("m1", """
                  <artifactId>m1</artifactId>
                  <version>1</version>
                  <parent>
                    <groupId>group</groupId>
                    <artifactId>parent</artifactId>
                  </parent>
                  """.trimIndent())
    maven.importProjectAsync()

    val action = RemoveManagedFilesAction()
    val parentActionVisible = action.isAvailable(TestActionEvent.createTestEvent(action, maven.createTestDataContext(parentFile)))
    val m1ActionVisible = action.isAvailable(TestActionEvent.createTestEvent(action, maven.createTestDataContext(m1File)))

    assertTrue(parentActionVisible)
    assertFalse(m1ActionVisible)
  }
}
